src/components/WorkspaceFilter/WorkspaceFilter.js

```diff
@@ -18,142 +19,75 @@ const WorkspaceFilter = props => {
   }
 
-  const toggleShow = () => setShow(!show)
-  // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
   const handleSelect = event => {
     event.persist()
-    switch(event.target.name) {
-      // Venue
-      case 'cafe':
-        setFilter(filter => ({...filter, cafe : !filter.cafe }))
-        break
-      case 'cowork':
-        setFilter(filter => ({...filter, cowork : !filter.cowork }))
-        break
-      case 'restaurant':
-        setFilter(filter => ({...filter, restaurant : !filter.restaurant }))
-        break
-      case 'library':
-        setFilter(filter => ({...filter, library : !filter.library }))
-        break
-      // Refreshments
-      case 'alcohol':
-        setFilter(filter => ({...filter, bool_alcohol : !filter.bool_alcohol }))
-        break
-      case 'coffee':
-        setFilter(filter => ({...filter, bool_coffee : !filter.bool_coffee }))
-        break
-      case 'food':
-        setFilter(filter => ({...filter, bool_food : !filter.bool_food }))
-        break
-      // Amenities
-      case 'fastWifi':
-        setFilter(filter => ({...filter, fastWifi : !filter.fastWifi}))
-        break
-      case 'bathrooms':
-        setFilter(filter => ({...filter, bathrooms : !filter.bathrooms }))
-        break
-      case 'comfiness':
-        setFilter(filter => ({...filter, bool_seating : !filter.bool_seating }))
-        break
-      case 'goodForGroups':
-        setFilter(filter => ({...filter, bool_goodforgroup : !filter.bool_goodforgroup }))
-      break
-      case 'lotsOfSeats':
-        setFilter(filter => ({...filter, lotsOfSeats : !filter.lotsOfSeats }))
-        break
-      case 'outdoorSpace':
-        setFilter(filter => ({...filter, bool_outdoorspace : !filter.bool_outdoorspace }))
-        break
-      case 'outlets':
-        setFilter(filter => ({...filter, bool_outlet : !filter.bool_outlet }))
-        break
-      case 'pets':
-        setFilter(filter => ({...filter, bool_petfriendly : !filter.bool_petfriendly }))
-        break
-      // Noise
-      case 'quiet':
-        setFilter(filter => ({...filter, quiet : !filter.quiet }))
-        break
-      case 'lively':
-        setFilter(filter => ({...filter, lively : !filter.lively }))
-        break
-      //Hours
-      case 'openEarly':
-        setFilter(filter => ({...filter, openEarly : !filter.openEarly }))
-        break
-      case 'openLate':
-        setFilter(filter => ({...filter, openLate : !filter.openLate }))
-        break
-      default:
-        console.log('Failed to apply filters!')
-    }
+    setFilter(filter => ({...filter, [event.target.name] : event.target.value }))
   }
 
   const handleSubmit = () => {
     doFiltering()
-      .then(resolve => props.filterWorkspaces(resolve))
-      .then(resolve => handleClose())
-      .then(setRejection(false))
-      .catch(reject => handleReject())
+      .then(resolve => {
+        props.filterWorkspaces(resolve)
+        clearFilter()

```

Comment:
do we need to clear filters on success? I think it would be good to maintain the state so user can easily update.

Reply:
My two cents! I like the idea of not clearing the filters, unless specifically prompted to do so. As Alex mentioned, it would make it easy to  update your filter choices.